# Cálculo - Tarefa 02

1- Calcule de forma numérica a integral da função $x^3$ avaliada entre os pontos $1$ e $2$. Identifique o número de retângulos que você usou.  
2- Calcule essa área de forma analítica.

### Calcule de forma numérica a integral da função  𝑥3 avaliada entre os pontos  1 e 2. Identifique o número de retângulos que você usou.

In [10]:
import pandas as pd
import numpy as np

In [11]:
f = lambda x: x**3

a = 1  # Ponto inicial
b = 2  # Ponto final
n = 1000  # Número de retângulos (aumente para maior precisão)

delta_x = (b - a) / n  # Largura de cada retângulo

integral = 0

for i in range(n):
    x_i = a + i * delta_x
    integral += f(x_i + 0.5 * delta_x) * delta_x  # Ponto médio do retângulo

print("Valor da integral:", integral)
print("Número de retângulos usados:", n)


Valor da integral: 3.749999625
Número de retângulos usados: 1000


### Calcule essa área de forma analítica.

## $\int_{1}^{2} x^3 \, dx = \left[\frac{1}{4}x^4\right]_{1}^{2} = \left(\frac{1}{4} \cdot 2^4\right) - \left(\frac{1}{4} \cdot 1^4\right) = \frac{16}{4} - \frac{1}{4} = 4 - \frac{1}{4} = \frac{15}{4} = 3.75$